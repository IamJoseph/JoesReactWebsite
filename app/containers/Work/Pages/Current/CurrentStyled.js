import styled from 'styled-components';

export const ImgContainer = styled.div`
  cursor: pointer;
  margin: 2em 0;
  position: relative;
  &.clicked{
    background-color: #fff;
    bottom: 0;
    display: flex;
    height: 100%;
    justify-content: center;
    left: 0;
    margin: 0;
    position: fixed;
    right: 0;
    top: 0;
    width: 100%;
    img{
      margin-top: 85px;
      max-height: calc(100% - 85px);
      max-width: 90%;
      width: auto;
      &:nth-child(2){
        display: none;
      }
    }
  }
`;

export const AWrapper = styled.div`
  line-height: normal;
  margin: 1.5em 0;
  a {
    color: #6d8da7;
  }
`;

export const IMG = styled.img`
  max-width: 41.9em;
  width: 100%;
`;

export const ImgSmall = styled.img`
  max-width: 5em;
  position: absolute;
  top: -40px;
  right: -40px;
  width: 100%;
`;

export const ExtraText = styled.div`
  font-size: .9em;
  line-height: 2.2em;
`;
