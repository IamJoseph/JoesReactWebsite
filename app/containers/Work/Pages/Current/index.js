import React, { PureComponent } from 'react';
import { Container, AWrapper, UL, H2, LI } from '../PagesStyled';
import { ImgContainer, IMG, ImgSmall, ExtraText } from './CurrentStyled';
import test from '../../../../assets/test_output.png';
import score from '../../../../assets/100.png';
import toggle from '../toggleFullScreen';

export default class Projects extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      fullscreen: false,
    };
  }

  toggleFullScreen = toggle;

  render() {
    const { fullscreen } = this.state;
    return (
      <Container>
        <UL>
          <H2>
            Current Site
          </H2>
          <LI>
            Es6/esNExt / immutable.js / react / redux / redux-saga / react-router /  styled-components /
            jest / enzyme / reselect / helmet / font face observer / service workers!
          </LI>
          <AWrapper>
            <a rel="dns-prefetch" href="https://github.com/IamJoseph/JoesReactWebsite" target="_blank">
            App @ GitHub
          </a>
          </AWrapper>
          <div>
            Sporting features such as SEO friendly meta-data, offline first, and check out those test scores!
          </div>
          <ImgContainer
            onClick={() => this.toggleFullScreen('fullscreen')}
            className={fullscreen && 'clicked'}
          >
            <IMG src={test} alt="output of running jest / enzyme tests" />
            <ImgSmall src={score} alt="emoji for emphasis" />
          </ImgContainer>
          <ExtraText>
            You might think to yourself, why so much effort for a <em>simple</em> website?
            <br /><br />Take a look at the code on Github
            and you will understand, it&apos;s all about scalability! Scalability, performance, and best practices.
            The same stack and architecture are applicable for enterprise level applications. <br /><br />I also enjoy new technologies and
            feel appreciation for the efforts of the talented devs creating these wonderful packages; it&apos;d be a crime not
            to use them!
          </ExtraText>
        </UL>
      </Container>
    );
  }
}
