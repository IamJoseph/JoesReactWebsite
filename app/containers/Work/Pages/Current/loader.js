/**
 * Asynchronously loads the components for WorkPage
 */

import { errorLoading } from 'utils/asyncInjectors';

export default () => (cb) => {
  import('containers/Work/Pages/Current')
    .then(cb)
    .catch(errorLoading);
};
