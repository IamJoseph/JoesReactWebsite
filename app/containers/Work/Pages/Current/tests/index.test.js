import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Current from '../index';
import { ImgContainer } from '../CurrentStyled';

describe('<Current />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Current />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should toggle className clicked on image container when clicked', () => {
    const wrapper = shallow(<Current />);
    wrapper.find(ImgContainer).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
