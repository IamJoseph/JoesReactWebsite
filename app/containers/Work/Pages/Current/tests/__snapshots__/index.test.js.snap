// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Current /> should render correctly 1`] = `
<styled.section>
  <styled.ul>
    <styled.h2>
      Current Site
    </styled.h2>
    <styled.li>
      Es6/esNExt / immutable.js / react / redux / redux-saga / react-router /  styled-components / jest / enzyme / reselect / helmet / font face observer / service workers!
    </styled.li>
    <styled.div>
      <a
        href="https://github.com/IamJoseph/JoesReactWebsite"
        rel="dns-prefetch"
        target="_blank"
      >
        App @ Github
      </a>
    </styled.div>
    <div>
      Sporting features such as SEO friendly meta-data, offline first, and check out those test scores!
    </div>
    <styled.div
      className={false}
      onClick={[Function]}
    >
      <styled.img
        alt="output of running jest / enzyme tests"
        src="IMAGE_MOCK"
      />
      <styled.img
        alt="emoji for emphasis"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div>
      You might think to yourself, why so much effort for a 
      <i>
        simple
      </i>
       website?
      <br />
      <br />
      Take a look at the code on Github and you will understand, it's all about scalability! Scalability, performance, and best practices. The same stack and architecture are applicable for enterprise level applications. 
      <br />
      <br />
      I also enjoy new technologies and feel appreciation for the efforts of the talented devs creating these wonderful packages; it'd be a crime not to use them!
    </styled.div>
  </styled.ul>
</styled.section>
`;

exports[`<Current /> should toggle className clicked on image container when clicked 1`] = `
<styled.section>
  <styled.ul>
    <styled.h2>
      Current Site
    </styled.h2>
    <styled.li>
      Es6/esNExt / immutable.js / react / redux / redux-saga / react-router /  styled-components / jest / enzyme / reselect / helmet / font face observer / service workers!
    </styled.li>
    <styled.div>
      <a
        href="https://github.com/IamJoseph/JoesReactWebsite"
        rel="dns-prefetch"
        target="_blank"
      >
        App @ Github
      </a>
    </styled.div>
    <div>
      Sporting features such as SEO friendly meta-data, offline first, and check out those test scores!
    </div>
    <styled.div
      className="clicked"
      onClick={[Function]}
    >
      <styled.img
        alt="output of running jest / enzyme tests"
        src="IMAGE_MOCK"
      />
      <styled.img
        alt="emoji for emphasis"
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.div>
      You might think to yourself, why so much effort for a 
      <i>
        simple
      </i>
       website?
      <br />
      <br />
      Take a look at the code on Github and you will understand, it's all about scalability! Scalability, performance, and best practices. The same stack and architecture are applicable for enterprise level applications. 
      <br />
      <br />
      I also enjoy new technologies and feel appreciation for the efforts of the talented devs creating these wonderful packages; it'd be a crime not to use them!
    </styled.div>
  </styled.ul>
</styled.section>
`;
